### PR TITLE
Add Phase-2 Level-1 LLP trigger emulator: TOoLLiP [14_0_0_pre3]

### PR DESCRIPTION
**Pull request by ddiaz006**
#### PR description:
* rebasing of https://github.com/cms-l1t-offline/cmssw/pull/1195
* Add ML-based LLP Jet tagger to Phase-2 Level-1 trigger emulator; codenamed TOoLLiP
* Requires TOoLLiP external integrated here: https://github.com/cms-sw/cmsdist/pull/8906
* Compiled object file added under L1Trigger/Phase2L1ParticleFlow/test until above PR is approved
* Presented at L1 Tau-Jets-MET meetings, e.g. 
  * https://indico.cern.ch/event/1288096/#9-llp-jet-tagging-status-and-p
  * https://indico.cern.ch/event/1350143/contributions/5683985/
* This model represents a v1 baseline with post-training quantization. We expect to update it with a more realistic firmware implementation.
* Model and expected performance plots shown here: https://github.com/cms-hls4ml/TOoLLiP/blob/main/README.md
* Central CMSSW  PR: https://github.com/cms-sw/cmssw/pull/43638

#### PR validation:

In progress

### CODE REVIEW

**Comment by ddiaz006** [comment]
To continue from the previous PR. Here I

- followed @aloeliger suggestions to remove some unnecessary comments.
- added a python config per @EmyrClement
- added the tagger result to the L1Emulator chain/event content
- removed the compiled object and squashed the commits
- also followed up with a study requested by @EmyrClement re the change from BJetID to JetID, shown in a recent Tau-Jets-Met meeting https://indico.cern.ch/event/1381574/#21-phase-2-llp-jet-tagging

**Comment by ddiaz006** [comment]
Re Emyr's comment in the last PR: https://github.com/cms-l1t-offline/cmssw/pull/1195#issuecomment-1948939573

I have changed the type so that the code does not crash and I checked edmDumpEventContent to verify that the LLPJets value maps are being saved.
```
edm::ValueMap<float>                  "l1tTOoLLiPProducer"        "L1PFLLPJets"     "L1P2GT"   
edm::ValueMap<float>                  "l1tTOoLLiPProducerCorrectedEmulator"   "L1PFLLPJets"     "L1P2GT" 
```

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 2 files that did not meet formatting requirements:
 - L1Trigger/L1TMuon/src/MicroGMTConfiguration.cc
 - L1Trigger/Phase2L1ParticleFlow/plugins/TOoLLiPProducer.cc

Please run `scram b code-format` to auto-apply code formatting
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

**Comment by epalencia** [comment]
@ddiaz006, could you please fix the code format in `L1Trigger/Phase2L1ParticleFlow/plugins/TOoLLiPProducer.cc`? you can forget about the other one.

File: L1Trigger/Phase2L1ParticleFlow/src/JetId.cc [hunk 1]
```diff
@@ -19,8 +40,9 @@ BJetId::BJetId(const std::string &iInput, const std::string &iOutput, const BJet
   fOutput_ = iOutput;
 }
 
-BJetId::~BJetId() {}
-void BJetId::setNNVectorVar() {
+JetId::~JetId() {}
```
Comment (aloeliger) [inline L1Trigger/Phase2L1ParticleFlow/src/JetId.cc]: Could you remove this and set the destructor to default in the class specification? After that is done, I think that everything seems in order here at least from me. @epalencia any objection to merging this into the prototype? If not @ddiaz006 I believe there is already a PR open for this in CMSSW, could you just make sure it is up to date with any changes you have made in review here?

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 1 files that did not meet formatting requirements:
 - L1Trigger/L1TMuon/src/MicroGMTConfiguration.cc

Please run `scram b code-format` to auto-apply code formatting
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

**Comment by epalencia** [comment]
@aloeliger, i'm fine with merging this PR. 

@ddiaz006, could you please port all latest commits to https://github.com/cms-sw/cmssw/pull/43638?

**Comment by ddiaz006** [comment]
Thanks, I am trying to update the PR to the main cms-sw, however it appears that I have pulled In some additional changes, how should I proceed? Should I rebase, if so to which target?

**Comment by aloeliger** [comment]
@ddiaz006 To provide the menu team with validation material, we are going to go ahead and merge this branch into the integration branch. If there are any elements of this that haven't been put in a PR to central CMSSW, please make a PR for that now. Also, I have left some instructions on the main CMSSW PR to help work out the extra commits. @epalencia FYI.

**Pull request merged**